File: test/fixtures/lang/sample.ts
// Hand-written TS fixture for the bake-off (NOT wh code). Public-domain, synthetic.
import { readFile } from "node:fs/promises";
import { helperUtil } from "./helpers";

export const MAX_RETRIES = 3;

export interface Repo {
  id: string;
  name: string;
}

/** Fetch a repo by id, with a bounded retry. */
export async function loadRepo(id: string, retries: number = MAX_RETRIES): Promise<Repo> {
  const raw = await readFile(id, "utf8");
  const parsed = helperUtil(raw);
  if (!parsed) throw new Error("bad repo");
  return { id, name: parsed };
}

export class RepoStore {
  private items: Map<string, Repo> = new Map();

  add(repo: Repo): void {
    this.items.set(repo.id, repo);
  }

  get(id: string): Repo | undefined {
    return this.items.get(id);
  }
}

function slugify(name: string): string {
  return name.toLowerCase().replace(/\s+/g, "-");
}

File: test/fixtures/lang/sample.cs
// Hand-written C# fixture for the bake-off (NOT wh code). Synthetic.
using System;
using System.Collections.Generic;
using System.Threading.Tasks;

namespace Acme.Repos
{
    public enum RepoKind
    {
        Git,
        Mercurial,
        Subversion
    }

    public interface IRepoStore
    {
        Repo Get(string id);
    }

    /// <summary>A repository record.</summary>
    public class Repo
    {
        public string Id { get; set; }
        public string Name { get; set; }
    }

    public class RepoStore : IRepoStore
    {
        private class CacheEntry
        {
            public Repo Value { get; set; }
        }

        private readonly Dictionary<string, Repo> _items = new();

        /// <summary>Adds a repository to the store.</summary>
        public void Add(Repo repo)
        {
            _items[repo.Id] = repo;
        }

        public Repo Get(string id)
        {
            return _items.TryGetValue(id, out var r) ? r : null;
        }

        public async Task<Repo> LoadAsync(string id)
        {
            var raw = await ReadAsync(id);
            return new Repo { Id = id, Name = Slugify(raw) };
        }

        private static string Slugify(string name)
        {
            return name.ToLowerInvariant().Replace(" ", "-");
        }

        private Task<string> ReadAsync(string id) => Task.FromResult(id);
    }
}
